pkg/services/ngalert/schedule/schedule.go

```diff
@@ -501,26 +434,15 @@ func (sch *schedule) stopApplied(alertDefKey ngmodels.AlertRuleKey) {
 	sch.stopAppliedFunc(alertDefKey)
 }
 
-func (sch *schedule) getRuleExtraLabels(ctx context.Context, alertRule *ngmodels.AlertRule) (map[string]string, error) {
+func (sch *schedule) getRuleExtraLabels(evalCtx *evaluation) map[string]string {
 	extraLabels := make(map[string]string, 4)
 
-	extraLabels[ngmodels.NamespaceUIDLabel] = alertRule.NamespaceUID
-	extraLabels[prometheusModel.AlertNameLabel] = alertRule.Title
-	extraLabels[ngmodels.RuleUIDLabel] = alertRule.UID
+	extraLabels[ngmodels.NamespaceUIDLabel] = evalCtx.rule.NamespaceUID
+	extraLabels[prometheusModel.AlertNameLabel] = evalCtx.rule.Title
+	extraLabels[ngmodels.RuleUIDLabel] = evalCtx.rule.UID
 
-	user := &models.SignedInUser{
-		UserId:  0,
-		OrgRole: models.ROLE_ADMIN,
-		OrgId:   alertRule.OrgID,
-	}
-
-	if !sch.disableGrafanaFolder {
-		folder, err := sch.ruleStore.GetNamespaceByUID(ctx, alertRule.NamespaceUID, alertRule.OrgID, user)
```

Comment:
this is the main point of this change. Prevent individual eval loop from fetching folder on its own.